tests/src/common/lease.rs

```diff
@@ -0,0 +1,382 @@
+    pub liability_init_percent: Percent,
+    pub liability_delta_to_healthy_percent: Percent,
+    pub liability_delta_to_max_percent: Percent,
+    pub liability_minus_delta_to_first_liq_warn: Percent,
+    pub liability_minus_delta_to_second_liq_warn: Percent,
+    pub liability_minus_delta_to_third_liq_warn: Percent,
+    pub liability_recalc_time: Duration,
+    // LoanForm
+    pub annual_margin_interest: Percent,
+    pub interest_payment: InterestPaymentSpec,
+    // Dex
+    pub dex: ConnectionParams,
+}
+
+impl Default for InstantiatorConfig {
+    fn default() -> Self {
+        Self {
+            customer: Addr::unchecked(USER),
+            liability_init_percent: Percent::from_percent(65),
+            liability_delta_to_healthy_percent: Percent::from_percent(5),
+            liability_delta_to_max_percent: Percent::from_percent(10),
+            liability_minus_delta_to_first_liq_warn: Percent::from_percent(2),
+            liability_minus_delta_to_second_liq_warn: Percent::from_percent(3),
+            liability_minus_delta_to_third_liq_warn: Percent::from_percent(2),
+            liability_recalc_time: Duration::from_days(20),
+
+            annual_margin_interest: Percent::from_percent(0), // 3.1%
+            interest_payment: InterestPaymentSpec::new(
+                Duration::from_secs(100),
+                Duration::from_secs(10),
+            ),
+
+            dex: ConnectionParams {
+                connection_id: "connection-0".into(),
+                transfer_channel: Ics20Channel {
+                    local_endpoint: "channel-0".into(),
+                    remote_endpoint: "channel-2048".into(),
+                },
+            },
+        }
+    }
+}
+
+#[derive(Debug, Clone)]
+pub struct InstantiatorAddresses {
+    pub lpp: Addr,
+    pub time_alarms: Addr,
+    pub oracle: Addr,
+    pub profit: Addr,
+}
+
+pub(crate) fn complete_lease_initialization<Lpn, DownpaymentC, LeaseC>(
+    app: &mut App,
+    connection_id: &str,
+    lease_addr: &Addr,
+    downpayment: Coin<DownpaymentC>,
+    exp_borrow: Coin<Lpn>,
+    exp_lease: Coin<LeaseC>,
+) where
+    Lpn: Currency,
+    DownpaymentC: Currency,
+    LeaseC: Currency,
+{
+    check_state_opening(app, lease_addr);
+
+    let ica_addr = "ica0";
+    let ica_port = format!("icacontroller-{ica_addr}");
+    let ica_channel = format!("channel-{ica_addr}");
+
+    let mut response: ResponseWithInterChainMsgs<'_, ()> = open_ica(
+        app,
+        lease_addr,
+        connection_id,
+        &ica_channel,
+        &ica_port,
+        ica_addr,
+    )
+    .ignore_result();
+
+    response.expect_ibc_transfer(
+        TestCase::LEASER_IBC_CHANNEL,
+        cwcoin::<DownpaymentC, _>(downpayment),
+        lease_addr.as_str(),
+        ica_addr,
+    );
+
+    () = response.unwrap_response();
+
+    check_state_opening(app, lease_addr);
+
+    let mut response: ResponseWithInterChainMsgs<'_, ()> =
+        send_blank_response(app, lease_addr).ignore_result();
```

Comment:
in an attempt to reduce this function length, extract this into a new fn, and reuse twice